tests/test_query.py

```diff
@@ -0,0 +1,90 @@
+from marshmallow import fields, Schema
+import pytest
+
+from flask_resty import Api, ApiView
+from flask_resty.testing import assert_response
+
+# -----------------------------------------------------------------------------
+
+
+@pytest.fixture
+def schemas():
+    class NameSchema(Schema):
+        name = fields.String(required=True)
+
+    class NameListSchema(Schema):
+        names = fields.List(fields.String(), load_from='name', required=True)
+
+    return {
+        'name': NameSchema(),
+        'name_list': NameListSchema(),
+    }
+
+
+@pytest.fixture(autouse=True)
+def routes(app, schemas):
+    class NameView(ApiView):
+        query_schema = schemas['name']
+
+        def get(self):
+            return self.make_response(self.get_query_data()['name'])
+
+    class NameListView(ApiView):
+        query_schema = schemas['name_list']
+
+        def get(self):
+            return self.make_response(self.get_query_data()['names'])
+
+    api = Api(app)
+    api.add_resource('/name', NameView)
+    api.add_resource('/names', NameListView)
+
+
+# -----------------------------------------------------------------------------
+
+
+def test_get_name_one(client):
+    response = client.get('/name?name=foo')
+    assert_response(response, 200, 'foo')
+
+
+def test_get_name_extra(client):
+    response = client.get('/name?name=foo&ignored=bar')
+    assert_response(response, 200, 'foo')
+
+
+def test_get_names_one(client):
+    response = client.get('/names?name=foo')
+    assert_response(response, 200, ['foo'])
+
+
+def test_get_names_many(client):
+    response = client.get('/names?name=foo&name=bar')
+    assert_response(response, 200, ['foo', 'bar'])
+
+
+# -----------------------------------------------------------------------------
+
+
+def test_error_get_name_missing(client):
+    response = client.get('/name')
+    assert_response(response, 422, [{
+        'code': 'invalid_query',
+        'source': {'parameter': 'name'},
+    }])
+
+
+def test_error_get_name_many(client):
```

Comment:
This is not the default behavior from Flask when reading query args, but I think it's more correct to fail on requests that specify a query parameter more than once when not expected.